inc/classes/class-srm-post-type.php

```diff
@@ -448,9 +492,78 @@ public function action_save_post( $post_id ) {
 			 * redirect info is saved, updating the cache before it is not sufficient.
 			 */
 			srm_flush_cache();
+
+			add_filter( 'redirect_post_location', array( $this, 'add_notice_query_var' ), 99, 2 );
 		}
 	}
 
+	/**
+	 * Add custom query var to indicate that this is a duplicate redirection rule.
+	 *
+	 * @param string $location The destination URL.
+	 * @param int    $post_id  The post ID.
+	 *
+	 * @return string
+	 */
+	public function add_notice_query_var( $location, $post_id ) {
+
+		remove_filter( 'redirect_post_location', array( $this, 'add_notice_query_var' ), 99 );
+
+		// If post ID is empty, bail out.
+		if ( empty( $post_id ) ) {
+			return $location;
+		}
+
+		$redirect_post = get_post( $post_id );
+
+		// If post not found or not of redirect_rule type, bail out.
+		if ( empty( $redirect_post ) || 'redirect_rule' !== $redirect_post->post_type ) {
+			return $location;
+		}
+
+		$redirect_from = get_post_meta( $post_id, '_redirect_rule_from', true );
+
+		if ( empty( $redirect_from ) ) {
+			return $location;
+		}
+
+		$existing_post_ids = new WP_Query(
+			[
+				'meta_key'       => '_redirect_rule_from',
+				'meta_value'     => $redirect_from,
+				'fields'         => 'ids',
+				'posts_per_page' => 1,
+				'no_found_rows'  => true,
+				'post_type'      => 'redirect_rule',
+				'post_status'    => 'publish',
+				'orderby'        => 'ID',
+				'order'          => 'ASC',
+			]
+		);
+
+		// If no posts found, then bail out.
+		if ( empty( $existing_post_ids->posts ) ) {
+			return $location;
+		}
+
+		$existing_post_id = $existing_post_ids->posts[0];
+
+		// If it is the same post ID, then bail out.
+		if ( (int) $post_id === (int) $existing_post_id ) {
+			return $location;
+		}
+
+		$location = add_query_arg(
+			array(
```

Comment:
I noticed you used `array()` here and `[]` above. Please use one format - whichever best matches what you see elsewhere in the codebase.